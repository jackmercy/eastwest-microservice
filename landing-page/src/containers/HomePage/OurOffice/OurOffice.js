import { memo, useMemo } from 'react';
import { useIntl } from 'react-intl';
import classNames from 'classnames';
import Link from 'next/link';
import { useMediaQuery } from 'react-responsive';

import { Content } from '../../components';
import BREAKPOINTS from '../../../constants/BREAKPOINTS';

import WebImages from './WebImages';
import MobileImages from './MobileImages';
import useStyles from './ourOffice.styles';

const OurOffice = () => {
  const intl = useIntl();
  const classes = useStyles();
  const isMobile = useMediaQuery({
    query: `(max-width: ${BREAKPOINTS.maxSm}px)`,
  });
  const images = useMemo(
    () => [
      {
        key: 'cover-1',
        src: '/static/images/homepage/office-01.png',
        srcSet: [
          '/static/images/homepage/office-01.png 500w',
          '/static/images/homepage/office-01@2x.png 1500w',
          '/static/images/homepage/office-01@3x.png 2000w',
        ].join(', '),
      },
      {
        key: 'cover-2',
        src: '/static/images/homepage/office-02.png',
        srcSet: [
          '/static/images/homepage/office-02.png 500w',
          '/static/images/homepage/office-02@2x.png 1500w',
          '/static/images/homepage/office-02@3x.png 2000w',
        ].join(', '),
      },
      {
        key: 'cover-3',
        src: '/static/images/homepage/office-03.png',
        srcSet: [
          '/static/images/homepage/office-03.png 500w',
          '/static/images/homepage/office-03@2x.png 1500w',
          '/static/images/homepage/office-03@3x.png 2000w',
        ].join(', '),
      },
      {
        key: 'cover-4',
        src: '/static/images/homepage/office-04.png',
        srcSet: [
          '/static/images/homepage/office-04.png 500w',
          '/static/images/homepage/office-04@2x.png 1500w',
          '/static/images/homepage/office-04@3x.png 2000w',
        ].join(', '),
      },
    ],
    [],
  );

  const ListImages = !isMobile ? WebImages : MobileImages;

  return (
    <Content className={classes.root}>
      <h2 className={classes.title} data-test="our-office-title">
        {intl.formatMessage({ id: 'HOMEPAGE.OUR_OFFICE.TITLE' })}
      </h2>
      <ListImages images={images} />
      <h2 className={classes.introTitle}>
        {intl.formatMessage({ id: 'HOMEPAGE.OUR_OFFICE.INTRO.TITLE.1' })}
      </h2>
      <p
        className={classes.introDescription}
        data-test="our-office-description"
      >
        {intl.formatMessage({
          id: 'HOMEPAGE.OUR_OFFICE.INTRO.DESCRIPTION.1',
        })}
      </p>
      <Link href="/talent-labs" passHref>
        <a
          href="/#"
          className={classNames(classes.introDescription, classes.introLink)}
          data-test="our-office-link-talent-labs"
        >
          {intl.formatMessage({ id: 'HOMEPAGE.OUR_OFFICE.LINK.1' })}
          <i className="icon-arrow-right" />
        </a>
      </Link>
      <h2 className={classes.introTitle}>
        {intl.formatMessage({ id: 'HOMEPAGE.OUR_OFFICE.INTRO.TITLE.2' })}
      </h2>
      <p
        className={classes.introDescription}
        data-test="our-office-description"
      >
        {intl.formatMessage({
          id: 'HOMEPAGE.OUR_OFFICE.INTRO.DESCRIPTION.2',
        })}
      </p>
      <Link href="/careers" passHref>
        <a
          href="/#"
          className={classNames(classes.introDescription, classes.introLink)}
          data-test="our-office-link-careers"
        >
          {intl.formatMessage({ id: 'HOMEPAGE.OUR_OFFICE.LINK.2' })}
          <i className="icon-arrow-right" />
        </a>
      </Link>
    </Content>
  );
};

export default memo(OurOffice);
