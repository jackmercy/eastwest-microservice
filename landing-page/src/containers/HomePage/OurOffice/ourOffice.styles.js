import { createUseStyles } from 'react-jss';

import { rem } from '../../../utils/jss';
import COLORS from '../../../constants/COLORS';
import BREAKPOINTS from '../../../constants/BREAKPOINTS';

export default createUseStyles({
  root: {
    overflow: 'hidden',
    maxWidth: rem(1192),
    paddingTop: rem(80),
    paddingBottom: rem(80),
  },
  title: {
    textAlign: 'center',
    fontWeight: 900,
    fontSize: rem(56),
    lineHeight: rem(72),
    transition: 'font-size .5s ease',
    marginTop: 0,
    marginBottom: rem(80),
  },
  introTitle: {
    width: '100%',
    fontWeight: 'bold',
    fontSize: rem(48),
    lineHeight: rem(64),
    marginTop: rem(48),
    marginBottom: 0,
    transition: 'font-size .5s ease, margin .5s ease',
  },
  introDescription: {
    fontSize: rem(32),
    lineHeight: rem(48),
    marginTop: rem(32),
    marginBottom: 0,
    transition: 'font-size .5s ease, margin .5s ease',
  },
  introLink: {
    display: 'inline-block',
    textDecoration: 'none',
    color: COLORS.primary,
    '& > [class^=icon-]': {
      fontSize: rem(11.5),
      verticalAlign: 'middle',
      marginLeft: rem(16),
    },
  },

  [`@media only screen and (max-width: ${BREAKPOINTS.maxSm}px)`]: {
    root: { paddingTop: rem(48), paddingBottom: rem(48) },
    title: { fontSize: rem(32), lineHeight: rem(40), marginBottom: rem(32) },
    introTitle: {
      fontSize: rem(24),
      lineHeight: rem(32),
      marginTop: rem(32),
      marginBottom: 0,
    },
    introDescription: {
      fontSize: rem(16),
      lineHeight: rem(24),
      marginTop: rem(16),
    },
  },
});
