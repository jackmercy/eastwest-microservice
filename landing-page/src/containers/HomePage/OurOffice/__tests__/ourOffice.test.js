import React from 'react';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import preloadAll from 'jest-next-dynamic';

import OurOffice from '../OurOffice';

describe('Testing HomePage > OurOffice', () => {
  let wrapper;

  beforeAll(async () => {
    // WebImages and MobileImages included in OurOffice
    // And they are using dynamic import, see more in WebImages.js and MobileImages.js
    // So we need wait for it to load all before we can assert anything
    await preloadAll();
    wrapper = mount(<OurOffice />);
  });

  it('should render match snapshot', () => {
    const wrapper = shallow(<OurOffice />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should have a title', () => {
    expect(wrapper.find('[data-test="our-office-title"]')).toHaveLength(1);
  });

  it('should have some descriptions', () => {
    expect(wrapper.find('[data-test="our-office-description"]')).toHaveLength(
      2,
    );
  });

  it('should have a link talent-labs', () => {
    expect(
      wrapper.find('[data-test="our-office-link-talent-labs"]'),
    ).toHaveLength(1);
  });

  it('should have a link careers', () => {
    expect(wrapper.find('[data-test="our-office-link-careers"]')).toHaveLength(
      1,
    );
  });
});
