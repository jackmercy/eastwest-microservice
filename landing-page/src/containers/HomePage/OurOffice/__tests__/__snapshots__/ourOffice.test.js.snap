// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing HomePage > OurOffice should render match snapshot 1`] = `
<Memo(Content)
  className="root-0-2-1"
>
  <h2
    className="title-0-2-2"
    data-test="our-office-title"
  >
    Our workspace in Vietnam
  </h2>
  <Memo(WebImages)
    images={
      Array [
        Object {
          "key": "cover-1",
          "src": "/static/images/homepage/office-01.png",
          "srcSet": "/static/images/homepage/office-01.png 500w, /static/images/homepage/office-01@2x.png 1500w, /static/images/homepage/office-01@3x.png 2000w",
        },
        Object {
          "key": "cover-2",
          "src": "/static/images/homepage/office-02.png",
          "srcSet": "/static/images/homepage/office-02.png 500w, /static/images/homepage/office-02@2x.png 1500w, /static/images/homepage/office-02@3x.png 2000w",
        },
        Object {
          "key": "cover-3",
          "src": "/static/images/homepage/office-03.png",
          "srcSet": "/static/images/homepage/office-03.png 500w, /static/images/homepage/office-03@2x.png 1500w, /static/images/homepage/office-03@3x.png 2000w",
        },
        Object {
          "key": "cover-4",
          "src": "/static/images/homepage/office-04.png",
          "srcSet": "/static/images/homepage/office-04.png 500w, /static/images/homepage/office-04@2x.png 1500w, /static/images/homepage/office-04@3x.png 2000w",
        },
      ]
    }
  />
  <h2
    className="introTitle-0-2-3"
  >
    We work with passion
  </h2>
  <p
    className="introDescription-0-2-4"
    data-test="our-office-description"
  >
    Our eyes are set beyond the world of engineering, we want to create an ecosytem anyone can use. We believe East West will be the go-to platform for excellence and product scalability.
  </p>
  <Link
    href="/talent-labs"
    passHref={true}
  >
    <a
      className="introDescription-0-2-4 introLink-0-2-5"
      data-test="our-office-link-talent-labs"
      href="/#"
    >
      Become a Talent at Talent Labs
      <i
        className="icon-arrow-right"
      />
    </a>
  </Link>
  <h2
    className="introTitle-0-2-3"
  >
    Help each other grow up
  </h2>
  <p
    className="introDescription-0-2-4"
    data-test="our-office-description"
  >
    Are you ready to innovate and make an impact? Join us, we are always looking for new talent! See all our current openings and apply to our vacancies and internships.
  </p>
  <Link
    href="/careers"
    passHref={true}
  >
    <a
      className="introDescription-0-2-4 introLink-0-2-5"
      data-test="our-office-link-careers"
      href="/#"
    >
      Learn more about opportunities at East West
      <i
        className="icon-arrow-right"
      />
    </a>
  </Link>
</Memo(Content)>
`;
